spec/enum_spec.rb

```diff
@@ -0,0 +1,197 @@
+    end
+
+    it 'Returns true if the block ever returns a value other than false or nil.' do
+      expect(animals.my_any? { |word| word == /d/ }).to be false
+    end
+
+    it 'Returns true if the block ever returns a value other than false or nil.' do
+      expect([nil, true, 99].my_any?(Integer)).to be true
+    end
+
+    it 'Returns true if the block ever returns a value other than false or nil.' do
+      expect([nil, true, 99].my_any?).to be true
+    end
+
+    it 'Returns true if the block ever returns a value other than false or nil.' do
+      expect([].my_any?).to be false
+    end
+  end
+
+  describe '#my_none?' do
+    it 'Returns true if the block never returns true for all elements.' do
+      expect(animals.my_none? { |word| word.length == 5 }).to be true
+    end
+
+    it 'Returns true if the block never returns true for all elements.' do
+      expect(animals.my_none? { |word| word.length >= 4 }).to be false
+    end
+
+    it 'Returns true if the block never returns true for all elements.' do
+      expect(animals.my_none? { |word| word == /d/ }).to be true
+    end
+
+    it 'Returns true if the block never returns true for all elements.' do
+      expect([1, 3.14, 42].my_none?(Float)).to be false
+    end
+
+    it 'Returns true if the block never returns true for all elements.' do
+      expect([].my_none?).to be true
+    end
+
+    it 'Returns true if the block never returns true for all elements.' do
+      expect([nil].my_none?).to be true
+    end
+
+    it 'Returns true if the block never returns true for all elements.' do
+      expect([nil, false].my_none?).to be true
+    end
+
+    it 'Returns true if the block never returns true for all elements.' do
+      expect([nil, false, true].my_none?).to be false
+    end
+  end
+
+  describe '#my_count' do
+    it 'Returns the number of items. If an argument is given, the number of items that are equal to the argument are counted' do
+      expect(array.my_count).to eql 4
+    end
+
+    it 'Returns the number of items. If an argument is given, the number of items that are equal to the argument are counted' do
+      expect(array.my_count(2)).to eql 2
+    end
+
+    it 'Returns the number of items. If an argument is given, the number of items that are equal to the argument are counted' do
+      expect(array.my_count(&:even?)).to eql 3
+    end
+  end
+
+  describe '#my_map' do
+    context 'When block given' do
+      it 'Returns a new array with the results of running a block once for every element.' do
+        expect(arr.my_map { |i| i * i }).to eql [1, 4, 9, 16, 25]
+      end
+
+      it 'Returns a new array with the results of running a block once for every element.' do
+        expect(arr.my_map { 'cat' }).to eql %w[cat cat cat cat cat]
+      end
+    end
+
+    context 'When block given' do
+      it 'Returns to enumerator' do
+        expect(arr.my_map).to be_an Enumerator
+      end
+    end
+  end
```

Comment:
- [ ]  Also the wording of the context description here is wrong, it should read `When no block is given`. Please fix it.